code/go/0chain.net/smartcontract/dbs/event/miner.go

```diff
@@ -49,19 +47,37 @@ func (edb *EventDb) GetMiner(id string) (Miner, error) {
 		First(&miner).Error
 }
 
-//func (edb *EventDb) MinerStats(id string) (*providerAggregateStats, error) {
-//	var miner providerAggregateStats
-//	result := edb.Store.Get().
-//		Model(&Miner{}).
-//		Where(&Miner{ID: id}).
-//		First(&miner)
-//	if result.Error != nil {
-//		return nil, fmt.Errorf("error retrieving miner %v, error %v",
-//			id, result.Error)
-//	}
-//
-//	return &miner, nil
-//}
+//nolint
+func (edb *EventDb) GetMinerWithDelegatePools(id string) (Miner, []DelegatePool, error) {
+	var minerDps []struct {
+		Miner
+		DelegatePool
+		ProviderRewards
+	}
+	var m Miner
+	var dps []DelegatePool
+
+	result := edb.Get().Preload("Rewards").
+		Table("miners").
+		Joins("left join provider_rewards on miners.id = provider_rewards.provider_id").
+		Joins("left join delegate_pools on miners.id = delegate_pools.provider_id").
```

Comment:
same as before, limit number of delegate pools

Reply:
As discussed, I added `include_delegates` option to /test/screst/nodeStat.